### PR TITLE
Upgrade JSDOM to v14

### PR DESCRIPTION
**Pull request by nktssh**
### 🔦 Summary
JSDom dependency was updated according to #1174 

### 🗺️ Test Plan
**TL;DR version**: We checked, there is v14! 😎

**Long version**: we checked it by creating additional test on our local env with one of the features were added to JSDOM since v11 (v13 actually, function `insertAdjacentElement ` is present as specified in changelog)

### CODE REVIEW

File: jest.unit.config.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
   testMatch: ['**/__tests__/**/*.spec.js'],
   testResultsProcessor: 'jest-teamcity-reporter',
+  testEnvironment: 'jest-environment-jsdom-fourteen',
```
Comment (ronami) [inline jest.unit.config.js]: I think this is not needed. We can probably even change this to `node`. This is the setup for Yoshi's unit tests (no need for `jsdom`).

File: package.json [hunk 2]
```diff
@@ -58,6 +58,7 @@
     "globby": "^9.0.0",
     "husky": "^1.0.0-rc.9",
     "jest": "^24.1.0",
+    "jest-environment-jsdom-fourteen": "^0.1.0",
```
Comment (ronami) [inline package.json]: Can probably be removed, see above.

File: packages/jest-yoshi-preset/jest-preset.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ module.exports = {
     ...[
       {
         displayName: 'spec',
-        testEnvironment: 'jsdom',
+        testEnvironment: 'jest-environment-jsdom-fourteen',
```
Comment (ronami) [inline packages/jest-yoshi-preset/jest-preset.js]: I think we should use `require.resolve()` just to make sure we don't get a different version of that package.

This is a breaking change and I want to test it in version 4 **before** we release the v5. What do you think about adding this behind a configuration option? We'll use `jest-environment-jsdom-fourteen` only if it's turned on (opt-in).

This way, we can merge and release this feature under v4. Projects can opt-into it by turning this config on. In v5 we can remove the opt-in and use this for everyone (breaking). 

cc @yavorsky @ranyitz

Comment (nktssh) [reply packages/jest-yoshi-preset/jest-preset.js]: @ronami we made environment variable for this

**Comment by nktssh** [comment]
@ronami is it good enough to merge?

**Pull request merged**